Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs

```diff
@@ -34,6 +34,444 @@ public void When_asserting_equality_with_a_collection_built_from_params_argument
+                DateTime.Now.AddDays(-1),
+                DateTime.Now.AddDays(2)
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action assert = () => actualDates.Should().BeEquivalentTo(expectedDates, (actual, expected) => actual.Should().BeSameDateAs(expected));
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Assert
+            //-----------------------------------------------------------------------------------------------------------
+            assert.ShouldNotThrow();
+        }
+
+        [TestMethod]
+        public void Check_asserting_equivallency_by_assert_method_should_check_it_when_objects_not_equal()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Assert
+            //-----------------------------------------------------------------------------------------------------------
+            List<DateTime> actualDates = new List<DateTime>
+            {
+                DateTime.Today,
+                DateTime.Today.AddDays(2),
+                DateTime.Today.AddDays(-1)
+            };
+
+            List<DateTime> expectedDates = new List<DateTime>
+            {
+                DateTime.Now,
+                DateTime.Now.AddDays(5),
+                DateTime.Now.AddDays(2)
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action assert = () =>
+            {
+                actualDates.Should()
+                    .BeEquivalentTo(expectedDates, (actual, expected) => actual.Should().BeSameDateAs(expected));
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Assert
+            //-----------------------------------------------------------------------------------------------------------
+            assert.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void Check_asserting_equivallency_by_assert_method_should_throw_exception_when_context_collection_is_bigger()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            List<DateTime> actualDates = new List<DateTime>
```

Comment:
Please replace some tests for datetimes with other types(ints, strings, custom classes and so on). Also, please provide tests for collections with null values.